Implement pgxn_meta CLI

theory opened the pull request:
Keep it simple right now, mimicking [validate_pgxn_meta], although the `--man` option currently doesn't do much. Use lexopt for argument parsing, since it's super simple like this use case.

Avoid [std::process::exit] and set a boolean to return [std::process::ExitCode] when the app should exit without processing a file. This makes it easier to test things, since the functions to print the usage statement or version don't exit.

Include tests that cover every bit of functionality, including the `main()` function. Add `corpus/invalid.json` to properly test validation failure. Try to improve the coveralls formatting.

  [validate_pgxn_meta]: https://metacpan.org/pod/validate_pgxn_meta
  [std::process::exit]: https://doc.rust-lang.org/stable/std/process/fn.exit.html
  [std::process::ExitCode]: https://doc.rust-lang.org/stable/std/process/struct.ExitCode.html

vrmiguel commented on src/main.rs:
```diff
@@ -0,0 +1,326 @@
+
+// Minimal main function; logical is all in run.
+fn main() -> Result<ExitCode, Box<dyn Error>> {
+    run(io::stdout(), env::args_os().collect::<Vec<_>>())
```
```suggestion
    run(io::stdout(), env::args_os())
```

Can likely get away from `collect`ing here since `args_os` already returns an iterator

theory replied:
Oh, thanks! Done in 9e072a1.

vrmiguel commented on src/main.rs:
```diff
@@ -0,0 +1,326 @@
+            let meta: Value = serde_json::from_reader(f)?;
+            let mut v = Validator::new();
+            if let Err(e) = v.validate(&meta) {
+                panic!("{file} {e}");
```
Why panicking instead of returning the error?

theory replied:
Because the error references values in meta, so can't be returned. It has since occurred to me that I could make a new error from a string and return that, but saw no urgency in it.

theory replied:
Made that change in 6fb932d

theory committed:
Implement pgxn_meta CLI

Keep it simple right now, mimicking [validate_pgxn_meta], although the
`--man` option currently doesn't do much. Use lexopt for argument
parsing, since it's super simple like this use case.

Avoid [std::process::exit] and set a boolean to return
[std::process::ExitCode] when the app should exit without processing a
file. This makes it easier to test things, since the functions to print
the usage statement or version don't exit.

Include tests that cover every bit of functionality, including the
`main()` function. Add `corpus/invalid.json` to properly test validation
failure. Try to improve the coveralls formatting.

  [validate_pgxn_meta]: https://metacpan.org/pod/validate_pgxn_meta
  [std::process::exit]: https://doc.rust-lang.org/stable/std/process/fn.exit.html
  [std::process::ExitCode]: https://doc.rust-lang.org/stable/std/process/struct.ExitCode.html
